deploy/supervisor/config.supervisor.pinniped.dev_oidcclients.yaml

```diff
@@ -119,6 +119,28 @@ spec:
                 minItems: 1
                 type: array
                 x-kubernetes-list-type: set
+              tokenLifetimes:
+                description: tokenLifetimes are the optional overrides of token lifetimes
+                  for an OIDCClient.
+                properties:
+                  idTokenSeconds:
+                    description: |-
+                      idTokenSeconds is the lifetime of ID tokens issued to this client, in seconds. This will choose the lifetime of
+                      ID tokens returned by the authorization flow and the refresh grant. It will not influence the lifetime of the ID
+                      tokens returned by RFC8693 token exchange. When null, a short-lived default value of approximately several
+                      minutes will be used. This value must be between 30 and 1,800 seconds (30 minutes), inclusive. It is recommended
+                      to make these tokens short-lived to force the client to perform the refresh grant often, because the refresh
+                      grant will check with the external identity provider to decide if it is acceptable for the end user to continue
```

Comment:
Does every refresh grant perform this external IDP check? Or is it rate-limited on the client (Pinniped) side?

Reply:
Every refresh grant goes back to the external IDP to perform some checks. Which specific checks are dependent on the IDP type and configuration.

Reply:
Maybe we should do some client-side rate limiting, so that this doesn't happen more than once every five minutes or so.

Reply:
The Pinniped CLI won't do it more often than every 5 minutes in the normal Supervisor + Concierge setup, because it will only do a refresh when both the mTLS client cert is expired (takes 5 minutes), and when the other tokens are also expired (takes less time).

Reply:
Right but `OIDCClients` will be able to have a token lifetime as low as 30 seconds, which could put a lot of additional burden on upstream refreshes.

Reply:
Hmm maybe that's a reason to make the minimum allowed value bigger? Like maybe make it 2 minutes?

Reply:
Refresh grants perform bcrypt operations, which are quite slow, so I think this is a good reason to make the minimum allowed value 2 minutes. Then we don't need to worry about architecting some kind of new rate limiting solution.

Reply:
Makes sense. We've never gotten a request for _shorter_ token lifetimes in any case.